src/domain/entities/DataSet.ts

```diff
@@ -39,6 +41,11 @@ export type CoreCompetency = { id: Id; name: string; code: string };
 export type DataSetList = Pick<DataSetAttrs, "id" | "name" | "lastUpdated" | "permissions">;
 
 export class DataSet extends Struct<DataSetAttrs>() {
+    validate(): Either<ValidationError<DataSet>[], DataSet> {
+        const allErrors = this.getValidationErrors();
+        return allErrors.length === 0 ? Either.success(this) : Either.error(allErrors);
+    }
```

Comment:
just as a note, an Either here is no strictly necessary, the return type could be `ValidationError<DataSet>[] | undefined`. But it's subjective if this is more declarative.